frontend/test/pytest/test_conditionals.py

```diff
@@ -200,6 +202,11 @@ def cond_fn():
         ):
             qjit(qml.qnode(qml.device(backend, wires=1))(circuit))
 
+        with pytest.raises(
+            TypeError, match="Conditional requires consistent return types across all branches"
+        ):
+            qjit(circuit)
```

Comment:
I believe we should, because we actually have different implementations for classical and quantum tracing. See [here](https://github.com/PennyLaneAI/catalyst/blob/2c8ce077743dd586025c759250ada01a57759835/frontend/catalyst/pennylane_extensions.py#L1363) vs [here](https://github.com/PennyLaneAI/catalyst/blob/2c8ce077743dd586025c759250ada01a57759835/frontend/catalyst/pennylane_extensions.py#L1398)

Reply:
Sounds good 👍